### PR TITLE
Bump deployment target to iOS 12

### LINKED ISSUE #83: Xcode 12 Support
Xcode 12 drops support for iOS 8. Release a new version that targets iOS 9+.

From the [Xcode 12 Beta Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-12-beta-release-notes):

> Xcode 12 beta 5 includes SDKs for iOS 14, iPadOS 14, tvOS 14, watchOS 7, and macOS 11. The Xcode 12 beta 5 release supports on-device debugging for iOS 9 and later, tvOS 9 and later, and watchOS 2 and later. Xcode 12 beta 5 requires Apple silicon running macOS Big Sur 11 beta or later, or an Intel-based Mac running macOS Catalina 10.15.4 or later.

### PR DESCRIPTION
**Pull request by NickEntin**
* Bump deployment target to iOS 12.0 and watchOS 4.0
* Bump minimum support Xcode version to 11.0
* Bump Swift version to 5.0
* Adopt recommended project settings
* Migrate off of deprecated APIs

Some commits in here are cherry picked off of @JustinDSN's #82 and rebased on top of the Aardvark 4.0 development branch. Resolves #83.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,8 +102,8 @@ Want to log with Aardvark but don’t want to use Aardvark’s bug reporting too
 
 ## Requirements
 
-* Xcode 8.0 or later
-* iOS 8 or later
+* Xcode 11.0 or later
+* iOS 12.0 or later
```
Comment (dfed) [inline README.md:106]: Fixing my past mistake:
```suggestion
* iOS 12.0 or later
* watchOS 4.0 or later
```

File: Sources/Aardvark/UIApplication+ARKAdditions.swift [hunk 2]
```diff
@@ -47,13 +47,8 @@ extension UIApplication {
         UIApplication.bugReporterToGestureRecognizerMap.setObject(bugReportingGestureRecognizer, forKey: bugReporter)
         
         if !UIApplication.observingKeyWindowNotifications {
-            #if swift(>=4.2)
```
Comment (dfed) [inline Sources/Aardvark/UIApplication+ARKAdditions.swift:50]: yay!

File: Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m [hunk 3]
```diff
@@ -27,6 +27,12 @@
 #import "ARKScreenshotViewController.h"
 #import "UIActivityViewController+ARKAdditions.h"
 
+typedef NS_ENUM(NSUInteger, ARKLogExportOption) {
```
Comment (dfed) [inline Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:30]: can you help me understand why this change was needed? This code looks reasonable but I'm not following why this was required.

File: Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m [hunk 4]
```diff
@@ -126,61 +125,18 @@ - (void)viewDidLoad;
...
     if (self.title.length == 0) {
         self.title = NSLocalizedString(@"Logs", @"Title of log viewing screen.");
     }
 }
 
-#pragma mark - UIActionSheetDelegate
-
-- (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex;
-{
-#if TARGET_IPHONE_SIMULATOR
```
Comment (dfed) [inline Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:147]: looks like this code block was moved down to `_exportLogsWithOption` with no meaningful changes

File: Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m [hunk 5]
```diff
@@ -126,61 +125,18 @@ - (void)viewDidLoad;
...
-            }
-        }
-        
-        return;
-    }
-#endif
-    
-    if (actionSheet == self.clearLogsConfirmationActionSheet) {
-        if (buttonIndex == actionSheet.destructiveButtonIndex) {
-            [self.logStore clearLogsWithCompletionHandler:NULL];
```
Comment (dfed) [inline Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:171]: this code moved down into an `UIAlertAction` handler with no changes

File: Sources/AardvarkMailUI/ARKEmailBugReporter.h [hunk 6]
```diff
@@ -100,7 +100,7 @@ typedef void (^ARKEmailBugReporterCustomPromptCompletionBlock)(ARKEmailBugReport
 /// Controls the number of recent error logs per log distributor to include in the email body of a bug report composed in a mail client that does not allow attachments. Defaults to 15.
 @property (nonatomic) NSUInteger numberOfRecentErrorLogsToIncludeInEmailBodyWhenAttachmentsAreUnavailable;
 
-/// The window level for the email composer on iOS 7 or later. Defaults to UIWindowLevelStatusBar + 3.0.
+/// The window level for the email composer. Defaults to UIWindowLevelStatusBar + 3.0.
```
Comment (dfed) [inline Sources/AardvarkMailUI/ARKEmailBugReporter.h:103]: the `+ 3` hack continues to give me joy. So many years later and it still works.

File: Sources/AardvarkMailUI/ARKEmailBugReporter.m [hunk 7]
```diff
@@ -493,17 +493,9 @@ - (void)_dismissEmailComposeWindow;
     
     [self.mailComposeViewController endAppearanceTransition];
     
-    static BOOL iOS9OrLater;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        iOS9OrLater = [[UIDevice currentDevice].systemVersion compare:@"9.0" options:NSNumericSearch] != NSOrderedAscending;
-    });
-    
     // Work around an iOS 9 bug where we don't get UIWindowDidBecomeKeyNotification when the mail compose view controller dismisses.
```
Comment (dfed) [inline Sources/AardvarkMailUI/ARKEmailBugReporter.m:496]: While it might be good to see if this bug still exists, let's make it a bit more clear why we're doing this:
```suggestion
    // Work around a bug introduced in iOS 9 where we don't get UIWindowDidBecomeKeyNotification when the mail compose view controller dismisses.
```

File: Sources/CoreAardvark/Logging/ARKDataArchive.m [hunk 8]
```diff
@@ -92,14 +92,10 @@ - (nullable instancetype)initWithURL:(nonnull NSURL *)fileURL maximumObjectCount
...
-    @try {
-        data = [NSKeyedArchiver archivedDataWithRootObject:object];
-    }
-    @catch (NSException *exception) {
-        ARKCheckCondition(NO, , @"Couldn't archive object %@", object);
-    }
+    NSError *error = nil;
+    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:object requiringSecureCoding:NO error:&error];
+
+    ARKCheckCondition(error == nil, , @"Couldn't archive object %@", object);
```
Comment (dfed) [inline Sources/CoreAardvark/Logging/ARKDataArchive.m:98]: nice update

File: Sources/CoreAardvark/Logging/ARKDataArchive.m [hunk 9]
```diff
@@ -143,7 +139,10 @@ - (void)readObjectsFromArchiveWithCompletionHandler:(nonnull void (^)(NSArray *
                 
                 id object = nil;
                 @try {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated"
```
Comment (dfed) [inline Sources/CoreAardvark/Logging/ARKDataArchive.m:143]: 😬 file an issue?

**Review by dfed — APPROVED** [review]
LGTM!

File: Sources/AardvarkMailUI/ARKEmailBugReporter.m (hunk 7 above)
Comment (NickEntin) [reply Sources/AardvarkMailUI/ARKEmailBugReporter.m:496]: Yeah, good call

File: Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m (hunk 3 above)
Comment (NickEntin) [reply Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:30]: This is one of @JustinDSN's commits, so he could explain the thinking behind the change better than I can. The short version is that `UIActionSheet` and `UIPopoverController` are deprecated in iOS 8.3 and iOS 9, respectively, so this replaces their usage with `UIAlertController`s.

This specific change (the enum of export options) could be implemented as two separate methods fairly easily. This route adds a bit more structure in order to avoid repeating a bit of logic. I don't have strong feelings either way given it's not a ton of structure but also not a ton of repeated logic.

File: Sources/CoreAardvark/Logging/ARKDataArchive.m (hunk 9 above)
Comment (NickEntin) [reply Sources/CoreAardvark/Logging/ARKDataArchive.m:143]: Filed as #93

File: Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m (hunk 3 above)
Comment (JustinDSN) [reply Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:30]: You summarized the specific trade offs I was trying to balance here. It's really good feedback to see the feedback on repeated logic and that would have been acceptable. Thanks for representing the thought process as well as seeing both sides of the same coin.

Comment (dfed) [reply Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:30]: A comment explaining the why would be helpful for future maintainers 🙂

Comment (JustinDSN) [reply Sources/AardvarkLoggingUI/Log Viewing/ARKLogTableViewController.m:30]: Done here: https://github.com/square/Aardvark/pull/94
